Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+import { useState } from 'react';
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    Tooltip,
+    Typography,
+} from '@material-ui/core';
+import {
+    ReferenceField,
+    ReferenceManyField,
+    SingleFieldList,
+} from 'react-admin';
+import get from 'lodash/get';
+import ChipConditionalLabel from '../../components/ChipConditionalLabel';
+import MappingButton from '../../components/MappingButton';
+import CollapseButton from '../../components/CollapseButton';
+
+const get_output_caps = (output_item, io) => {
+    return (
+        <>
+            <Typography variant="body1">{'Description:'}</Typography>
+            {output_item.properties.description}
+            <Typography variant="body1">{'Routable inputs:'}</Typography>
+            {output_item.caps.routable_inputs !== null
+                ? output_item.caps.routable_inputs
+                      .map(input_name => {
+                          return input_name === null
+                              ? 'MUTE'
+                              : get(io, `inputs.${input_name}.properties.name`);
+                      })
+                      .join(', ')
+                : 'null'}
+        </>
+    );
+};
+
+const get_input_caps = input_item => {
+    return (
+        <>
+            <Typography variant="body1">{'Description:'}</Typography>
+            {get(input_item, `properties.description`)}
+            <Typography variant="body1">{'Block size:'}</Typography>
+            {get(input_item, `caps.block_size`)}
+            <Typography variant="body1">{'Reordering:'}</Typography>
+            {String(get(input_item, `caps.reordering`))}
+        </>
+    );
+};
+
+const get_source_tooltip = output_item => {
+    return (
+        <>
+            <Typography variant="body1">{'Flows:'} </Typography>
+            <div>
+                <ReferenceManyField
+                    record={output_item}
+                    basePath="/flows"
+                    label="Flows"
+                    source="source_id"
+                    reference="flows"
+                    target="source_id"
+                    link="show"
+                >
+                    <div
+                        style={{
+                            margin: 5,
+                            padding: 10,
+                        }}
+                    >
+                        <SingleFieldList linkType="show">
+                            <ChipConditionalLabel source="label" />
+                        </SingleFieldList>
+                    </div>
+                </ReferenceManyField>
+            </div>
+            <Typography variant="body1">{'Senders:'} </Typography>
+            <div
+                style={{
+                    margin: 2,
+                    padding: 1,
+                }}
+            >
+                <ReferenceManyField
```

Comment:
I wonder how many API calls the nested `ReferenceManyField`s result in, and whether it's worth trying to be more efficient.

Reply:
Looked good on the call today! Worth keeping an eye on the browser developer tools console and network tabs though, to keep the number of API calls and console logging minimized.

I will investigate whether the `paging.limit=1000` that @rhastie spotted will cause any problems.